test/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+var agent = require('../../..').start({
+  serviceName: 'test',
+  secretToken: 'test',
+  captureExceptions: false,
+  metricsInterval: 0
+})
+
+var test = require('tape')
+var mockClient = require('../../_mock_http_client')
+test(function (t) {
+  resetAgent(function (data) {
+    t.end()
+  })
+  var Memcached = require('memcached')
+  var cache = new Memcached(`127.0.0.1:11211`, { timeout: 500 })
+  agent.startTransaction('foo', 'bar')
+  cache.set('foo', 'bar', 300, (err) => {
+    t.error(err)
+    cache.get('foo', (err, data) => {
+      t.error(err)
+      t.equal(data, 'bar')
+      cache.replace('foo', 'fizz', 300, (err) => {
+        t.error(err)
+        cache.get('foo', (err, data) => {
+          t.error(err)
+          t.equal(data, 'fizz')
+          cache.touch('foo', 300, (err) => {
+            t.error(err)
+            cache.del('foo', (err) => {
+              t.error(err)
+              cache.get('foo', (err, data) => {
+                t.error(err)
+                t.equal(data, undefined)
+                agent.endTransaction()
+                agent.flush()
+                t.end()
```

Comment:
Just remove this line as the test should not end until the `resetAgent` callback is called (within which you need to check that the `data` is as expected).